aws/runner.go

```diff
@@ -1,41 +1,63 @@
 package aws
 
 import (
+	"fmt"
 	"log"
 
 	"github.com/hashicorp/hcl/v2"
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
 	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
 )
 
 // Runner is a wrapper of RPC client for inserting custom actions for AWS provider.
 type Runner struct {
 	tflint.Runner
 	PluginConfig *Config
-	AwsClient    *Client
+	AwsClients   map[string]*Client
 }
 
 // NewRunner returns a custom AWS runner.
 func NewRunner(runner tflint.Runner, config *Config) (*Runner, error) {
-	var client *Client
+	clients := map[string]*Client{
+		"aws": nil,
+	}
 	if config.DeepCheck {
 		credentials, err := GetCredentialsFromProvider(runner)
 		if err != nil {
 			return nil, err
 		}
-
-		client, err = NewClient(config.toCredentials().Merge(credentials))
-		if err != nil {
-			return nil, err
+		for k, cred := range credentials {
+			client, err := NewClient(config.toCredentials().Merge(cred))
+			if err != nil {
+				return nil, err
+			}
+			clients[k] = client
 		}
 	}
 
 	return &Runner{
 		Runner:       runner,
 		PluginConfig: config,
-		AwsClient:    client,
+		AwsClients:   clients,
 	}, nil
 }
 
+func (r *Runner) AwsClient(attributes hclext.Attributes) (*Client, error) {
+	provider := "aws"
+	if attr, exists := attributes["provider"]; exists {
+		if err := r.EnsureNoError(r.EvaluateExpr(attr.Expr, &provider, nil), func() error { return nil }); err != nil {
```

Comment:
It doesn't work well.

Reply:
You have to get the alias from the attribute.

e.g.

```tf
resource "aws_instance" "foo" {
  provider = aws.west # You have to get the alias "west".

  # ...
}
```

Reply:
To support Terraform Modules, probably you have to fix something, but unfortunately I don't know well.

```tf
module "aws_vpc" {
  source = "./aws_vpc"
  providers = {
    aws = aws.west
  }
}
```

Reply:
> To support Terraform Modules, probably you have to fix something, but unfortunately I don't know well.

Umm, it's a hard problem. To solve this problem, we need to refer to the provider map of the module call.

The `Check()` method of each rule is called for each module (The root module is also one of the modules). So, `GetModuleContent` basically cannot access the parent module containing a module call that calls itself.

The easiest way is to stop support for alias in child modules. If you haven't faced this issue with aliases in modules, you'll be able to accept it.

The best way is to add an API that references the parent module. `GetModuleContent` has options for the module context, imagine something like `ParentModuleCtxType`.
https://github.com/terraform-linters/tflint-plugin-sdk/blob/v0.10.1/tflint/option.go#L18-L19

You should probably be able to achieve the intended behavior by getting the provider map with `Check()` method in the `RuleSet` and then caching it in the Runner.
https://github.com/terraform-linters/tflint-ruleset-aws/blob/7523ed68628e673e2e76572e32634fb5002b5285/aws/ruleset.go#L46-L59

Reply:
```tf
  provider      = aws.us-east-1
```

```
07:23:43 runner_eval.go:40: [INFO] unevaluable expression found in main.tf:27. TFLint ignores unevaluable expressions.
07:23:43 [ERROR] interceptor/logging.go:18: failed to gRPC request: direction=plugin2host method=/proto.Runner/EvaluateExpr err="rpc error: code = FailedPrecondition desc = unevaluable expression found in main.tf:27"
```

Hmm... It fails to parse the provider attribute.

Reply:
`EvaluateExpr` limits the types of expressions that can be evaluated, so cannot evaluate expressions like `aws.*`.
https://github.com/terraform-linters/tflint/blob/v0.35.0/tflint/runner_eval.go#L93-L104

In the Terraform core, it seems that alias doesn't support evaluation, just get it with `gohcl.DecodeExpression`. The same way can be used.
https://github.com/hashicorp/terraform/blob/v1.1.7/internal/configs/provider.go#L58

Reply:
Thank you for your help.

In the Terraform core, resource's "provider" attribute seems to be parsed here.
https://github.com/hashicorp/terraform/blob/3fbedf25430ead97eb42575d344427db3c32d524/internal/configs/resource.go#L136-L140